api/src/com/cloud/network/Networks.java

```diff
@@ -251,6 +252,10 @@ public static URI fromString(String candidate) {
                 if (com.cloud.dc.Vlan.UNTAGGED.equalsIgnoreCase(candidate)) {
                     return Native.toUri(candidate);
                 }
+                if (UuidUtils.validateUUID(candidate)){
+                    //Supporting lswitch uuid as vlan id: set broadcast_uri = null and add row on nicira_nvp_router_map with logicalrouter_uuid = candidate
+                    return null;
```

Comment:
I've read the comment in the line above a few times, and still cannot understand the purpose of returning null on an invalid UUID (i.e. you previously caught a number format exception).